scripts/us_bea/states_gdp/import_industry_data_and_gen_mcf.py

```diff
@@ -0,0 +1,168 @@
+             'New Hampshire', 'New Jersey', 'New Mexico', 'New York',
+             'North Carolina', 'North Dakota', 'Ohio', 'Oklahoma', 'Oregon',
+             'Pennsylvania', 'Rhode Island', 'South Carolina', 'South Dakota',
+             'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington',
+             'West Virginia', 'Wisconsin', 'Wyoming']
+
+# It would be great to make this class inherit from StateGDPDataLoader to
+# avoid repeating e.g. the dowload_data function. As soon as that gets merged
+# in, I will modify this to inherit from there.
+class StateGDPIndustryDataLoader:
+    """Pulls GDP industry-level state data from BEA.
+
+    Attributes:
+        df: DataFrame (DF) with the cleaned data.
+    """
+    ZIP_LINK = "https://apps.bea.gov/regional/zip/SQGDP.zip"
+    FILE = "SQGDP2__ALL_AREAS_2005_2019.csv"
+
+    def __init__(self):
+        """Downloads the data, cleans it and stores it in instance DF."""
+        df = self.download_data()
+
+        # Filters out columns that are not US States (e.g. New England).
+        # TODO(fpernice): Add non-state entities.
+        df = df[df['GeoName'].isin(US_STATES)]
+
+        # Gets columns that represent quarters, e.g. 2015:Q2, by matching
+        # against a regular expression.
+        all_quarters = [q for q in df.columns if re.match(r"....:Q.", q)]
+
+        # Convert table from wide to long format.
+        df = pd.melt(df, id_vars=['GeoFIPS', 'IndustryClassification'],
+                     value_vars=all_quarters,
+                     var_name='Quarter')
+
+        qtr_month_map = {
+            'Q1':'03',
+            'Q2':'06',
+            'Q3':'09',
+            'Q4':'12'
+        }
+
+        def date_to_obs_date(date):
+            """Converts date format e.g. 2005:Q3 to e.g. 2005-09."""
+            return date[:4] + "-" + qtr_month_map[date[5:]]
+        df['Quarter'] = df['Quarter'].apply(date_to_obs_date)
+
+        def convert_geoid(fips_code):
+            """Creates GeoId column. We get lucky that Data Commons's geoIds
+            equal US FIPS state codes.
+            """
+            fips_code = fips_code.replace('"', "")
+            fips_code = fips_code.replace(" ", "")
+            return "geoId/" + fips_code[:2]
+        df['GeoId'] = df['GeoFIPS'].apply(convert_geoid)
+
+        df = df[df['IndustryClassification'] != "..."]
+
+        def convert_industry_class(naics_code):
+            """Filters out aggregate NAICS codes and assigns them their Data
+            Commons codes.
+            """
+            if naics_code == "321,327-339":
+                return "JOLTS_320000"
+            if naics_code == "311-316,322-326":
+                return "JOLTS_340000"
+            return naics_code.replace("-", "_")
+        df['NAICS'] = df['IndustryClassification'].apply(convert_industry_class)
+
+        self.df = df.drop(["GeoFIPS","IndustryClassification"], axis=1)
+
+    def download_data(self):
```

Comment:
I think this refers only to class members, not the object's members.  https://realpython.com/instance-class-and-static-methods-demystified/ has an overview.  A class method can use only members for the entire class, not those of individual instances.

```
@classmethod
def download_data(cls):
```

Then replace all "self." by "cls.".

Reply:
Changed this function such that it now saves to instance df instead of returning. Thus is has to be instance method.